crates/atuin-daemon/src/server.rs

```diff
@@ -182,10 +184,40 @@ pub async fn listen(
         history_db,
     ));
 
-    Server::builder()
-        .add_service(HistoryServer::new(history))
-        .serve_with_incoming_shutdown(uds_stream, shutdown_signal(socket.into()))
-        .await?;
+    #[cfg(unix)]
```

Comment:
I think it would be nice to split these into two platform-specific functions, and then call them. The two blocks are rather large